glitch/analysis/security.py

```diff
@@ -1,15 +1,1107 @@
+                                        wildcarded_resource = True
+                                        break
+                                    i += 1
+                                    resource = self.check_required_attribute(statement.keyvalues, [""], f"resources[{i}]")
+                                if wildcarded_resource:
+                                    errors.append(Error('sec_sensitive_iam_action', resource, file, repr(resource)))
+                elif (element.type in ["resource.aws_iam_role_policy", "resource.aws_iam_policy", 
+                                       "resource.aws_iam_user_policy", "resource.aws_iam_group_policy"]):
+                    policy = self.check_required_attribute(element.attributes, [""], "policy")
+                    if policy:
+                        policy_dict = convert_string_to_dict(policy.value.lower())
+                        if policy_dict and policy_dict["statement"]:
+                            statements = policy_dict["statement"]
+                            if isinstance(statements, dict):
+                                statements = [statements]
+                            for statement in statements:
+                                if statement["effect"] and statement["action"] and statement["resource"]:
+                                    if statement["effect"] == "allow":
+                                        if isinstance(statement["action"], list):
+                                            for action in statement["action"]:
+                                                if ("*" in action):
+                                                    errors.append(Error('sec_sensitive_iam_action', policy, file, repr(policy)))
+                                                    break
+                                        else:
+                                            if ("*" in statement["action"]):
+                                                errors.append(Error('sec_sensitive_iam_action', policy, file, repr(policy)))
+                                        if isinstance(statement["resource"], list):
+                                            for resource in statement["resource"]:
+                                                if (resource in ["*"]) or (":*" in resource):
+                                                    errors.append(Error('sec_sensitive_iam_action', policy, file, repr(policy)))
+                                                    break
+                                        else:
+                                            if (statement["resource"] in ["*"]) or (":*" in statement["resource"]):
+                                                errors.append(Error('sec_sensitive_iam_action', policy, file, repr(policy)))
+            
+            elif isinstance(element, Attribute) or isinstance(element, Variable):
+                pass
```

Comment:
Remove?

Reply:
Yes it can be removed, there are other cases that this happened. I decided to keep the elif statement with the pass condition just to follow the structure of the terraform smell checkers. Since there are no values in the terraform config for this type of smell, this clause is empty. The analysis is performed only in the atomic unit.